crates/zksync/compiler/src/zksolc/compile.rs

```diff
@@ -71,16 +75,32 @@ pub type ContractBytecodes = BTreeMap<String, String>;
 
 #[derive(Debug, Default, Clone)]
 pub struct ZkSolcArtifactPaths {
-    artifact: PathBuf,
-    contract_hash: PathBuf,
+    base: PathBuf,
 }
 
 impl ZkSolcArtifactPaths {
     pub fn new(filename: PathBuf) -> Self {
-        Self {
-            artifact: filename.join("artifacts.json"),
-            contract_hash: filename.join("contract_hash"),
-        }
+        Self { base: filename }
```

Comment:
There's currently a major bug in how we cache and compile artifacts from solc. I'm working on fixing that and aligning the artifact generation (and caching). Can we limit this PR to only the necessary modifications - or perhaps rebase it after the bug is fixed?

Reply:
We can merge/rebase it after. The process doesn't depend much on the precise way we store artifacts